### PR TITLE
SYN-339: Cancel the notifier timeout when the notifier fires

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: synapse/notifier.py [hunk 1]
```diff
@@ -93,6 +93,13 @@ def notify(self, notifier, events, start_token, end_token):
                 self.appservice, set()
             ).discard(self)
 
+        # Cancel the timeout for this notifer if one exists.
+        if self.timer is not None:
+            try:
+                notifier.clock.cancel_call_later(self.timer)
+            except:
+                logger.exception("Failed to cancel notifier timer")
```
Comment (erikjohnston) [inline synapse/notifier.py]: Will `cancel_call_later` no raise if the timer had already been cancelled? The timeout function invokes this method.

Comment (erikjohnston) [inline synapse/notifier.py]: http://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IDelayedCall.html

Comment (NegativeMjark) [inline synapse/notifier.py]: Hmm, looks like cancel_call_later just calls IDelayedCall.cancel. Which as you point out will throw if the timer has already fired.

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**